bin/phpsdk_setshell.bat

```diff
@@ -81,21 +81,24 @@ if 15 gtr %PHP_SDK_VS_NUM% (
 	)
 	for /f "tokens=2*" %%a in ('reg query !TMPKEY! /v ProductDir') do set PHP_SDK_VC_DIR=%%b
 ) else (
-	rem the first one found seems the correct one
-	for /f "tokens=1* delims=: " %%a in ('%~dp0\vswhere -nologo -version %PHP_SDK_VS_NUM% -requires Microsoft.VisualStudio.Component.VC.Tools.x86.x64 -property installationPath -format text') do (
+	rem build the version range, e.g. "[15,16)"
+	set /a PHP_SDK_VS_RANGE=PHP_SDK_VS_NUM + 1
+	set PHP_SDK_VS_RANGE="[%PHP_SDK_VS_NUM%,!PHP_SDK_VS_RANGE%!)"
+
+	for /f "tokens=1* delims=: " %%a in ('%~dp0\vswhere -nologo -version !PHP_SDK_VS_RANGE! -requires Microsoft.VisualStudio.Component.VC.Tools.x86.x64 -property installationPath -format text') do (

```

Comment:
Great idea with the range. With this, the breaks are not needed in this loop, as it delivers just one result. Perhaps it would make sense to get rid of these breaks? The reason I rewrote it is that on my side the order was reversed - first 2017, then 2019. Now with your approach it seems to be consistent.

Thanks.

Reply:
Ah, yeah. I wonder what would happen if one has a preview and an RC installed side-by-side, though. In that case likely two installations will be found, i.e. two lines.

Reply:
Nope, preview is checked further down the line, attaching `-prerelease` to the command. Until `vswhere` changes behaviors, it should be ok.

Thanks.